keen/api.py

```diff
@@ -179,13 +167,14 @@ def delete_events(self, event_collection, params):
         self._error_handling(response)
         return True
 
+    @requires_key(KeenKeys.READ)
```

Comment:
Note that the change to use `read_key` for get_collection() and get_all_collections() is intentional. The SDK has been incorrectly requiring the `master_key`, but we should use the key of least privilege.